feat(details): view cost model details from price list modal

boaz0 opened the pull request:
closes cost 255

dlabrecq commented on src/pages/details/components/priceList/priceListModal.tsx:
```diff
@@ -74,12 +76,17 @@ class PriceListModalBase extends React.Component<Props> {
     return priceListRates ? (
-      <PriceListTable t={t} rates={priceListRates} />
+      <>
+        <PriceListTable t={t} rates={priceListRates} />
+        <Link to={`cost-models/${priceListRates.uuid}`}>
+          {t('details.price_list.modal.cost_model_link')}
+        </Link>
+      </>
```
Perhaps you and Natalie discussed an alternate design, but don't believe we wanted to add a link in the existing price list table?

Instead of showing a price list, I believe the intention was that the kebab would take the user directly to the cost models details page.

> Desired Functionality: When a user is looking at a list of clusters or cloud accounts, we have an option in the row kebab to redirect the user to the assigned cost model's details page (if one is assigned to it).  If there is no associated source id with a cost model, we can have the option ... disabled.

boaz0 replied:
@dlabrecq I recall that you said it will be odd to have that option in the kebab because the user excepts a modal and not to be redirected to the cost model page.
@nlcwong do you remember what we decided to do here?

dlabrecq commented:
When clicking on the "View cost model" link, I see the "page not found" screen. The URL in the browser is "undefined". For example `https://ci.foo.redhat.com:1337/beta/cost-management/details/cost-models/undefined`.

And for AWS, I get the "no rates found" screen. I didn't think we supported price lists for AWS at this time?

boaz0 commented on src/pages/details/awsDetails/detailsTable.tsx:
```diff
@@ -154,46 +155,51 @@ class DetailsTableBase extends React.Component<DetailsTableProps> {
-          title: '',
-        },
-      ];
+    const columns =
+      groupByTagKey || groupByOrg
+        ? [
+            {
+              title: t('ocp_details.tag_column_title'),
+            },
+            {
+              title: t('aws_details.change_column_title'),
+            },
+            {
+              orderBy: 'cost',
+              title: t('aws_details.cost_column_title', { total }),
+              transforms: [sortable],
+            },
+            {
+              title: '',
+            },
+          ]
+        : [
+            {
+              orderBy: groupById === 'account' ? 'account_alias' : groupById,
+              title: t('aws_details.name_column_title', { groupBy: groupById }),
+              transforms: [sortable],
+            },
+            {
+              title: t('aws_details.change_column_title'),
+            },
+            {
+              orderBy: 'cost',
+              title: t('aws_details.cost_column_title'),
+              transforms: [sortable],
+            },
+            {
+              title: '',
+            },
+          ];
 
```
You tell me, this is a change in `package.json` that caused this odd behavior :man_shrugging:

boaz0 commented on src/pages/details/awsDetails/detailsTable.tsx:
```diff
@@ -264,6 +271,7 @@ class DetailsTableBase extends React.Component<DetailsTableProps> {
         item={item}
         query={query}
         reportPathsType={reportPathsType}
+        showPriceListOption
```
I remembered differently. AWS/Azure/OCP all have `sources` in the API

dlabrecq replied:
Yes, they all have sources. However, I don't believe AWS has a price list .

dlabrecq replied:
Not a big deal, just wondering why your environment is formatting differently than mine?

dlabrecq replied:
If there's no AWS price list, then users won't see the link to sources -- they will only see an empty state.

boaz0 commented on src/utils/computedReport/getComputedReportItems.ts:
```diff
@@ -96,7 +97,10 @@ export function getUnsortedComputedReportItems<
 
         // org_unit_id workaround for storage and instance-type APIs
-        const id = idKey === 'org_entities' ? value.id || value.org_unit_id : value[idKey];
+        const id =
+          idKey === 'org_entities'
+            ? value.id || value.org_unit_id
+            : value[idKey];
```
I have no idea. I have a feeling it's related to `prettier` and its configuration file. For some reason I do not see that configuration and I assumed it was removed for obvious reasons.

Now I noticed that `prettier` is not installed (it's not in `package.json`) - do you have an idea why since you did the last change there.

boaz0 commented on src/pages/details/components/priceList/priceListModal.tsx:
```diff
@@ -9,10 +9,12 @@ import { ErrorState } from 'components/state/errorState/errorState';
 import { priceListActions, priceListSelectors } from 'store/priceList';
 import { providersSelectors } from 'store/providers';
 import { ComputedReportItem } from 'utils/computedReport/getComputedReportItems';
+import { ReportPathsType } from '../../../../api/reports/report';
```
yeah sure

boaz0 commented on src/pages/details/components/priceList/priceListModal.tsx:
```diff
@@ -102,9 +109,9 @@ class PriceListModalBase extends React.Component<Props> {
 }
 
 const PriceListModal = connect(
-  createMapStateToProps((state, { item, providerType }) => {
+  createMapStateToProps((state, { item, reportPathsType }) => {
     let type;
-    switch (providerType) {
+    switch (reportPathsType) {
       case ProviderType.aws:
         type = 'AWS';
```
no problem

dlabrecq commented on src/store/costModels/actions.ts:
```diff
@@ -126,3 +128,32 @@ export const deleteCostModel = (
+    return apiGetCostModels(`source_uuid=${source_uuid}`).then(res => {
+      if(res.data.data.length === 0) {
+        dispatch(
+          addNotification({
+            title: i18next.t('cost_models_router.error_title'),
+            description: i18next.t('cost_models_router.no_match'),
+            variant: 'danger',
+            dismissable: true,
+          })
```
I don't understand why we need to dispatch a 'danger' notifications? 

Couldn't we disable the 'View all price lists' option in the kebab? The /reports/openshift/costs API already provides the `source_uuid` in the report. This information is already available to each row of the table.

<img width="541" alt="Screen Shot 2020-08-05 at 4 10 53 PM" src="https://user-images.githubusercontent.com/17481322/89459214-4a1f6a00-d736-11ea-85fe-1b9c4f42b541.png">

We should be able to perform a `history.push` via the kebab's `onClick` event.

boaz0 replied:
Done. :+1:

boaz0 replied:
Although, we do need this redux thunk because we need to get the cost model that this source uuid is asigned to.

boaz0 commented:
Hi @dlabrecq - I updated the code. Removed the unnecessary notification and disabled the "View price list" when there is no cost model. I also fixed the navigation problem.

boaz0 commented:
Thanks a lot @dlabrecq
